jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -6616,36 +6620,204 @@ private static char convEndian(boolean isBigEndian, char value) {
 	 */
 	public native long getObjectSize(Object o);
 
+	/**
+	 * Determines whether a field at a given offset in a given class is flattened
+	 *
+	 * @param clz the the class to check
+	 * @param offset offset in bytes of the field in clz
+	 * @return a boolean indicating whether a flattened field exists at the given offset
```

Comment:
Please add a comment saying it returns false on arrays.

Reply:
added in 7159341